webui/react/src/components/CheckpointDetails.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { PropsWithChildren, useCallback } from 'react';
+
+import useModalCheckpoint from 'hooks/useModal/useModalCheckpoint';
+import {
+  CheckpointDetail, CheckpointWorkloadExtended, ExperimentBase,
+} from 'types';
+
+interface Props {
+  checkpoint: CheckpointWorkloadExtended | CheckpointDetail;
+  experiment: ExperimentBase;
+  title: string;
+}
+
+const CheckpointDetails: React.FC<Props> = (
+  {
+    checkpoint,
+    experiment,
+    title,
+    ...props
+  }: PropsWithChildren<Props>,
+) => {
+  const { modalOpen: openModalDelete } =
+  useModalCheckpoint({ checkpoint: checkpoint, config: experiment.config, title: title });
+  const handleModalCheckpointClick = useCallback(() => openModalDelete(), [ openModalDelete ]);
+
+  return (
+    <span onClick={() => handleModalCheckpointClick()}>
+      {props.children}
+    </span>
+  );
+};
+
+export default CheckpointDetails;
```

Comment:
**blocking question:** so I'm not sure if this component is needed. What if you used the `useModalCheckpoint` directly where you need it?
```
const { modalOpen: openModalDelete } =
  useModalCheckpoint({ checkpoint: checkpoint, config: experiment.config, title: title });

// somewhere in the JSX
<SomeComponent onClick={openModalDelete} />
```

Reply:
Initially I had implemented the hook this way but the implementation did not work as intended due to how the `checkpoint` state is being set. For example, in the current `ExperimentTrials` the `activeCheckpoint` is set [as a result of a button click](https://github.com/determined-ai/determined/blob/ad06cd43145725c5f1ea52ed36254bd6035f5fb1/webui/react/src/pages/ExperimentDetails/ExperimentTrials.tsx#L52). 

Therefore on initial page load, using the modal such as `useModalCheckpoint({ checkpoint: checkpoint, config: experiment.config, title: title });` would initialize the hook with a `checkpoint` that was undefined. And clicking on the the button that would reveal the modal actual would not render as you would expect since the the `activeCheckpoint` would not be set until after the modal has shown, due to the state not having been updated by the time the hook was called. 

The other pages using the `CheckpointModal` follow a similar pattern, where the `checkpoint` being used would not necessarily be set by the time the `useModalCheckpoint` was called. The choice to use an extra component was made to resolve this issue. 

With the current implementation, the hook will always be called with the correct checkpoint since the `CheckpointDetails` will always be rendered with the appropriate `checkpoint`. 

Let me know if that explanation makes sense and sounds reasonable.

Reply:
Ah I see, so I guess `CheckpointDetails` acts as the modal trigger. So couple of things:

**suggestion:** This is more of a pattern discussion and FYI for future modals too. If there are things you want to pass to the hook but aren't read during hook definition, e.g. `checkpoint`, then you can move it from a prop over to `openOrUpdate` as a function param instead. So when you are ready to open the modal, the variable dependency will be available then.

**nit suggestion:** thoughts on renaming `CheckpointDetails` to be `CheckpointModalTrigger` to be a bit more specific for what it is doing?

Reply:
The naming update to `CheckpointModalTrigger` has been made.